src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -469,6 +469,19 @@ public void replace(CtElement element) {
 		ReplacementVisitor.replace(this, elements);
 	}
 
+	@Override
+	public <E extends CtElement> E setMetadata(Map<String, Object> metadata) {
```

Comment:
question for @monperrus and @pvojtechovsky : shouldn't we call the model change listener here? Do we consider that metadata are not part of the model and the change on them shouldn't be notified?

Reply:
I would say change listener doesn't have to call change listener. But if one will needed. It is no problem to implement it later.

Reply:
ok then I just wanted to be sure it was not something missing.